orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -185,6 +185,33 @@ static std::vector<ArgDef> AddViewForParameter(
   return view_outputs;
 }
 
+void PartitionOptimizerState(
+    const int64_t partition_offset,
+    const int64_t partition_size,
+    NameMLValMap& initial_states) {
+  const std::vector<std::string> moments_prefixes({"Moment_1", "Moment_2"});
+  for (const auto& moments_prefix : moments_prefixes) {
+    if (initial_states.find(moments_prefix) != initial_states.end()) {
+      const auto& initial_state_it = initial_states.find(moments_prefix);
+      auto* init_tensor = initial_state_it->second.GetMutable<Tensor>();
+      float* data_buffer = init_tensor->MutableData<float>();
```

Comment:
>float* data_buffer = init_tensor->MutableData<float>(); [](start = 6, length = 55)

so we are assuming the data type is the same size as a float. is that valid?

Reply:
changing to allow mlfloat16 as well and error out for any other type.